src/components/Header/Header.tsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { TodoContext } from '../Store/TodoContext';
+import { Todo } from '../../types/Todo';
+import { addTodos, updateTodos } from '../../api/todos';
+
+export const Header: React.FC = () => {
+  const {
+    setErrorMessage,
+    errorMessage,
+    setTodos,
+    todos,
+    setTempTodo,
+    tempTodo,
+  } = useContext(TodoContext);
+  const [title, setTitle] = useState('');
+  const focus = useRef<HTMLInputElement>(null);
+
+  const addTodo = (data: Todo) => {
+    addTodos(data)
+      .then(newTodo => {
+        setTitle('');
+        setTodos(currentTodo => [...currentTodo, newTodo]);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+      })
+      .finally(() => {
+        setTempTodo(null);
+      });
+  };
+
+  const creatNewTodo = () => {
+    const newTodo: Todo = {
+      id: 0,
+      userId: 260,
+      title,
+      completed: false,
+    };
+
+    setTempTodo(newTodo);
+    addTodo(newTodo);
+  };
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (title.length < 1 || title.trim() === '') {
+      setErrorMessage('Title should not be empty');
```

Comment:
make enum for all possible error messages. and use it here and in all other places.